.tools/validate.go

```diff
@@ -0,0 +1,269 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+// DefaultRules are the standard validation to perform on git commits
+var DefaultRules = []ValidateRule{
+	func(c CommitEntry) (vr ValidateResult, err error) {
+		vr.CommitEntry = c
+		hasValid := false
+		for _, line := range strings.Split(c.Body, "\n") {
+			if validDCO.MatchString(line) {
+				hasValid = true
+			}
+		}
+		if !hasValid {
+			vr.Pass = false
+			vr.Msg = "does not have a valid DCO"
+		} else {
+			vr.Pass = true
+			vr.Msg = "has a valid DCO"
+		}
+
+		return vr, nil
```

Comment:
If we never return anything other than `nil` for `err`, I'd just make the signature:

    func(c CommitEntry) (vr ValidateResult)